companion/lib/Surface/Controller.ts

```diff
@@ -1266,6 +1266,10 @@ export class SurfaceController extends EventEmitter<SurfaceControllerEvents> {
 		for (const [surfaceId, surfaceConfig] of Object.entries(surfaces)) {
 			const surface = this.#getSurfaceHandlerForId(surfaceId, true)
 			if (surface) {
+				// it appears that #surfaceHandlers and #surfaceGroups have independent copies of `groupConfig`...
+				//  and the one in #surfaceGroups is the one that controls the surface's ..page.. values.
+				const group = this.#surfaceGroups.get(surfaceId)
```

Comment:
should this be up here or below the `this.#attachSurfaceToGroup(surface)`, because that call can change whether this will find a group

Reply:
I suspect that if `this.#attachSurfaceToGroup(surface)` was called, then my fix wouldn't be necessary since it appears to recreate the whole object. Nevertheless, I agree that moving that  line is a good idea.
Done!